src/operator/numpy/random/np_uniform_op.h

```diff
@@ -0,0 +1,269 @@
+#define MXNET_OPERATOR_NUMPY_RANDOM_NP_UNIFORM_OP_H_
+
+#include <mxnet/operator_util.h>
+#include <mshadow/base.h>
+#include <vector>
+#include <string>
+#include <algorithm>
+#include "./twoparams_dist_common.h"
+#include "../../elemwise_op_common.h"
+#include "../../tensor/elemwise_binary_broadcast_op.h"
+#include "../../mshadow_op.h"
+#include "../../mxnet_op.h"
+#include "../../operator_common.h"
+
+namespace mxnet {
+namespace op {
+
+struct NumpyUniformParam : public dmlc::Parameter<NumpyUniformParam> {
+  int t;
+  float low;
+  float high;
+  std::string ctx;
+  int dtype;
+  dmlc::optional<mxnet::Tuple<int>> size;
+  DMLC_DECLARE_PARAMETER(NumpyUniformParam) {
+    DMLC_DECLARE_FIELD(low);
+    DMLC_DECLARE_FIELD(high);
+    DMLC_DECLARE_FIELD(t)
+        .describe("input type indicator, "
+                  "0: array array "
+                  "1: scalar array "
+                  "2: array scalar "
+                  "3: scalar scalar");
+    DMLC_DECLARE_FIELD(size)
+        .set_default(dmlc::optional<mxnet::Tuple<int>>())
+        .describe("Output shape. If the given shape is, "
+                  "e.g., (m, n, k), then m * n * k samples are drawn. "
+                  "Default is None, in which case a single value is returned.");
+    DMLC_DECLARE_FIELD(ctx)
+    .set_default("")
+    .describe("Context of output, in format [cpu|gpu|cpu_pinned](n)."
+              " Only used for imperative calls.");
+    DMLC_DECLARE_FIELD(dtype)
+    .add_enum("None", -1)
+    .add_enum("float32", mshadow::kFloat32)
+    .add_enum("float64", mshadow::kFloat64)
+    .add_enum("float16", mshadow::kFloat16)
+    .set_default(-1)
+    .describe("DType of the output in case this can't be inferred. "
+              "Defaults to float32 if not defined (dtype=None).");
+  }
+};
+
+
+inline bool NumpyUniformOpShape(const nnvm::NodeAttrs &attrs,
+                                std::vector<TShape> *in_attrs,
+                                std::vector<TShape> *out_attrs) {
+  const NumpyUniformParam &param = nnvm::get<NumpyUniformParam>(attrs.parsed);
+  CHECK_EQ(out_attrs->size(), 1U);
+  if (param.size.has_value()) {
+    // Size declared.
+    std::vector<dim_t> oshape_vec;
+    const mxnet::Tuple<int> &size = param.size.value();
+    for (int i = 0; i < size.ndim(); ++i) {
+      oshape_vec.emplace_back(size[i]);
+    }
+    SHAPE_ASSIGN_CHECK(*out_attrs, 0, TShape(oshape_vec));
+    for (size_t input_idx = 0; input_idx < in_attrs->size(); input_idx++) {
+      CheckBroadcastable((*in_attrs)[input_idx], (*out_attrs)[0]);
+    }
+  } else {
+    // Size undeclared.
+    if (in_attrs->size() == 2U) {
+      // Low and high both from ndarray.
+      mxnet::TShape& low = (*in_attrs)[0];
+      mxnet::TShape& high = (*in_attrs)[1];
+      mxnet::TShape out(std::max(low.ndim(), high.ndim()), -1);
+      InferBroadcastShape(low, high, &out);
+      SHAPE_ASSIGN_CHECK(*out_attrs, 0, out);
+    }
+    if (in_attrs->size() == 1U) {
```

Comment:
better to use "else if" so that it won't be evaluated if previous condition matches.

Reply:
> better to use "else if" so that it won't be evaluated if previous condition matches.

fixed, thx